server/channels/app/plugin_requests.go

```diff
@@ -158,96 +188,73 @@ func (ch *Channels) servePluginRequest(w http.ResponseWriter, r *http.Request, h
 	}
 
 	if mfaAppErr := app.MFARequired(rctx); mfaAppErr != nil {
-		pluginID := mux.Vars(r)["plugin_id"]
+		pluginID := params["plugin_id"]
 		ch.srv.Log().Warn("Treating session as unauthenticated since MFA required",
 			mlog.String("plugin_id", pluginID),
 			mlog.String("url", r.URL.Path),
 			mlog.Err(mfaAppErr),
 		)
-		token = ""
+		handler(context, w, r)
+		return
 	}
 
-	// Mattermost-Plugin-ID can only be set by inter-plugin requests
-	r.Header.Del("Mattermost-Plugin-ID")
+	csrfCheckPassed := false
+	if (session != nil && session.Id != "") && appErr == nil && cookieAuth && r.Method != "GET" {
+		sentToken := ""
 
-	r.Header.Del("Mattermost-User-Id")
-	if token != "" {
-		csrfCheckPassed := false
-		if (session != nil && session.Id != "") && appErr == nil && cookieAuth && r.Method != "GET" {
-			sentToken := ""
-
-			if r.Header.Get(model.HeaderCsrfToken) == "" {
-				bodyBytes, _ := io.ReadAll(r.Body)
-				r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
-				if err := r.ParseForm(); err != nil {
-					mlog.Warn("Failed to parse form data for plugin request", mlog.Err(err))
-				}
-				sentToken = r.FormValue("csrf")
-				r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
-			} else {
-				sentToken = r.Header.Get(model.HeaderCsrfToken)
+		if r.Header.Get(model.HeaderCsrfToken) == "" {
+			bodyBytes, _ := io.ReadAll(r.Body)
+			r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+			if err := r.ParseForm(); err != nil {
+				mlog.Warn("Failed to parse form data for plugin request", mlog.Err(err))
 			}
+			sentToken = r.FormValue("csrf")
+			r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+		} else {
+			sentToken = r.Header.Get(model.HeaderCsrfToken)
```

Comment:
Can we set this up front, then check if `sentToken == ""` and then attempt to read it as a form value?